sktree/tree/_unsup_criterion.pyx

```diff
@@ -299,16 +401,48 @@ cdef class TwoMeans(UnsupervisedCriterion):
                 w = self.sample_weight[s_idx]
 
             ss += w * (self.Xf[s_idx] - mean) * (self.Xf[s_idx] - mean)
+        # if ss == 0.0: ss = 1e-10
```

Comment:
Here is the LOC to address `int` and 'nan', but then again it breaks fastBIC, so commented out here

Reply:
This would just make getting zero variance impossible. Zero variance is well defined. 

Log(0) is what is not well defined.